enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -37,26 +38,787 @@ proposal's scope.
+  // configA and configB. The change in configA triggers the rollout. If the configB is also
+  // changed before the rollout is complete, the current rollout stops and the rollout starts over.
+  // +optional
+  RolloutStrategy RolloutStrategy `json:"rolloutStrategy,omitempty"`
+}
+
+// RolloutStrategy represents the rollout strategy of the add-on configuration.
+type RolloutStrategy struct {
+  // Type is the type of the rollout strategy, it supports UpdateAll and RollingUpdateWithPlacement:
+  // - UpdateAll: when configs change, apply the new configs to all the clusters.
+  // - RollingUpdateWithPlacement: when configs change, rolling update new configs on the clusters 
+  //   selected by placements. 
+  //   If any of the configs are overridden by ManagedClusterAddOn on the specific cluster, the new configs 
+  //   won't take effect on that cluster.
+  //   This rollout strategy is only responsible for applying new configs. When the strategy is modified or 
+  //   removed, the applied configs won't be deleted from the cluster.
+  // +kubebuilder:validation:Enum=RollingUpdateWithPlacement;UpdateAll
+  // +kubebuilder:default:=UpdateAll
+  // +optional
+  Type string `json:"type"`
+
+  // Rolling update with placement config params. Present only if the type is RollingUpdateWithPlacement.
+  // +optional
+  RollingUpdateWithPlacement *RollingUpdateWithPlacement `json:"rollingUpdateWithPlacement,omitempty"`
+}
+
+// RollingUpdateWithPlacement represents the placement and behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdateWithPlacement struct {
+  // name of the placement
+  // +kubebuilder:validation:Required
+  // +required
+  Name string `json:"name"`
+
+  // namespace of the placement.
+  // +kubebuilder:validation:Required
+  // +required
+  Namespace string `json:"namespace"`
+
+  // The maximum concurrently updating number of addons.
+  // Value can be an absolute number (ex: 5) or a percentage of desired addons (ex: 10%).
+  // Absolute number is calculated from percentage by rounding up.
+  // Defaults to 25%.
+  // Example: when this is set to 30%, once the addon configs change, the addon on 30% of the selected clusters
+  // will adopt the new configs. When the new configs are ready, the addon on the remaining clusters
+  // will be further updated.
+  // +optional
+  MaxConcurrentlyUpdating *intstr.IntOrString `json:"maxConcurrentlyUpdating,omitempty"`
+}
+```
+
+`ManagedClusterAddOnStatus` adds `SupportedConfigs` to list the configuration types which are allowed to override the 
+the configuration defined in `ClusterManagementAddOn`.
+
+`ManagedClusterAddOnStatus` remove `LastObservedGeneration` and use `DesiredConfigSpecHash` and `LastAppliedConfigSpecHash` 
+to record the configuration spec hash.
+
+```golang
+// ManagedClusterAddOnStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ManagedClusterAddOnStatus struct {
```

Comment:
When performing a canary rollout, you want new clusters that are added to the placement to get the old configuration, correct?  If so, you need to track that old configuration somehow.  It could be done in status or it could be done in a different resource.  I suspect status here will be easier, but I don't know for certain.

I think the configuration is logically the combination of *all* the ConfigReferences, not the individual parts.

Reply:
Some corner cases consideration is added to the proposal, including the newly added clusters.